graphql-dgs-spring-boot-micrometer/src/main/kotlin/com/netflix/graphql/dgs/metrics/micrometer/DgsGraphQLMetricsProperties.kt

```diff
@@ -1,15 +1,18 @@
 package com.netflix.graphql.dgs.metrics.micrometer
 
 import org.springframework.boot.actuate.autoconfigure.metrics.AutoTimeProperties
+import org.springframework.boot.actuate.autoconfigure.metrics.PropertiesAutoTimer
 import org.springframework.boot.context.properties.ConfigurationProperties
 import org.springframework.boot.context.properties.NestedConfigurationProperty
 import org.springframework.boot.context.properties.bind.DefaultValue
 
 @ConfigurationProperties("management.metrics.dgs-graphql")
 data class DgsGraphQLMetricsProperties(
     /** Auto-timed queries settings. */
+    var autotimeProperties: AutoTimeProperties = AutoTimeProperties(),
+    /** Auto-timer. */
     @NestedConfigurationProperty
-    var autotime: AutoTimeProperties = AutoTimeProperties(),
+    var autotime: PropertiesAutoTimer = PropertiesAutoTimer(autotimeProperties),
```

Comment:
This one probably needs some input from @srinivasankavitha & @berngp.

This change wasn't well documented and I had to do some digging to figure out why this change was made. Spring Boot RC1 attempts to decouple Micrometer from `MetricsProperties` and `MetricsProperties` references `AutoTimeProperties`. Seems reasonable...

See spring boot change here: https://github.com/spring-projects/spring-boot/commit/180d0edbca63d0d28903b94b0f20f5f1e20a0c7f

I composed the properties and timer in this data class to provide easy access to `AutoTimeProperties`, but very open to different approaches / ideas.